### PR TITLE
[civ2][flakiness/2] Migrate core medium python tests to civ02

### PR DESCRIPTION
**Pull request by can-anyscale**
Migrate core medium python tests to civ2. Concretely
- Splitting the existing medium step to run non-core tests + flaky-core tests
- Move non-flaky-core tests to civ2
- Also remove the 'kubernetes' tag filters since this tag no longer exist

Tests:
- CI

### CODE REVIEW

**Review by aslonnie — COMMENTED** [review]
seems to depend on the 1st part?

File: .buildkite/pipeline.build_core.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #ci:group=:book: core tests (civ2)
 
 - label: ":python: (small & client)"
-  conditions: ["RAY_CI_PYTHON_AFFECTED"]
```
Comment (aslonnie) [inline .buildkite/pipeline.build_core.yml:4]: why is this line removed?

File: .buildkite/pipeline.build_core.yml [hunk 2]
```diff
@@ -12,3 +11,21 @@
...
       -- $(bazel query "attr(tags, team:core, tests(python/ray/tests/...))") $(yq -r .flaky_tests[] ci/v2/test/core.tests.yml | sed -e 's/^/-/')
+
+- label: ":python: (medium a-j)"
+  instance_size: medium
+  commands:
+    - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
+    - ./ci/env/env_info.sh
+    - bazel test --config=ci $(./ci/run/bazel_export_options)
+      --test_tag_filters=medium_size_python_tests_a_to_j
+      -- $(bazel query "attr(tags, team:core, tests(python/ray/tests/...))") $(yq -r .flaky_tests[] ci/v2/test/core.tests.yml | sed -e 's/^/-/')
```
Comment (aslonnie) [inline .buildkite/pipeline.build_core.yml:22]: try with no sed?

File: .buildkite/pipeline.build_core.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/pipeline.build_core.yml]: i was removing it for ci to trigger the job for testing, added back

**Review by aslonnie — APPROVED** [review]
lgtm, not sure if you want some one from core team to have a look too.

**Pull request merged**